### PR TITLE
Turn on some automation for the new cluster-api-provider-gcp repository.

### PR DESCRIPTION
**Pull request by roberthbailey**
Once this merges and squash-merges https://github.com/kubernetes-sigs/cluster-api-provider-gcp/pull/6 I'll send a follow-up PR to actually run some real tests.

### CODE REVIEW

File: prow/config.yaml [hunk 1]
```diff
@@ -227,6 +227,7 @@ tide:
     - kubernetes/website
     - kubernetes-sigs/application
     - kubernetes-sigs/cluster-api
+    - kubernetes-sigs/cluster-api-provider-gcp
```
Comment (BenTheElder) [inline prow/config.yaml]: before we enable merge automation we also need to at least enable in `prow/plugins.yaml`: `approve` for this repo so `/approve` works, this merge query requires certain labels (see line 239 here)

Comment (roberthbailey) [reply prow/config.yaml]: I've added the config to `prow/plugins.yaml` and squashed it into this PR.

**Pull request merged**